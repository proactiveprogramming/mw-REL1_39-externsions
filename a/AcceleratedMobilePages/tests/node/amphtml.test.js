var ampHtmlValidator = require('amphtml-validator');
var fs = require('fs');
var path = require('path');

function logAmpFailures(result) {
	for (var ii = 0; ii < result.errors.length; ii++) {
		var error = result.errors[ii];
		var msg = 'line ' + error.line + ', col ' + error.col + ': ' + error.message;
		if (error.specUrl !== null) {
			msg += ' (see ' + error.specUrl + ')';
		}
		(error.severity === 'ERROR' ? console.error : console.warn)(msg);
	}
}

describe('ampHtml', function () {
	var ampHtml;

	beforeAll(function () {
		ampHtml = fs.readFileSync(path.join(__dirname, '../amp.html'), "utf8");
	});

	it('is valid AMP HTML', function (done) {
		ampHtmlValidator.getInstance().then(function (validator) {
			var result = validator.validateString(ampHtml);
			try {
				expect(result.status).toBe('PASS');
			} catch (e) {
				logAmpFailures(result);
				done(e);
				return;
			}
			done();
		});
	});

	it('matches snapshot', function () {
		expect(ampHtml).toMatchSnapshot();
	});
});
