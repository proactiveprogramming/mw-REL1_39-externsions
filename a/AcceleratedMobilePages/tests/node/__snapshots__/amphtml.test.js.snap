// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ampHtml matches snapshot 1`] = `
"<!doctype html>
<html âš¡ lang=\\"en\\">
<head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width,minimum-scale=1\\">
	<meta name=\\"description\\" content=\\"\\">
    <link rel=\\"preload\\" as=\\"script\\" href=\\"https://cdn.ampproject.org/v0.js\\">
    <script async src=\\"https://cdn.ampproject.org/v0.js\\"></script>
    <script async custom-element=\\"amp-auto-ads\\" src=\\"https://cdn.ampproject.org/v0/amp-auto-ads-0.1.js\\"></script>
    <script async custom-element=\\"amp-analytics\\" src=\\"https://cdn.ampproject.org/v0/amp-analytics-0.1.js\\"></script>
	<script async custom-element=\\"amp-form\\" src=\\"https://cdn.ampproject.org/v0/amp-form-0.1.js\\"></script>
    <style amp-custom>
        html{font-family:sans-serif;font-size:0.875em;line-height:1.6}body{margin:0}header{display:flex;justify-content:space-between;align-items:center;background-color:#eaecf0;padding:0 1.5em;height:64px;border:0;box-shadow:inset 0 -1px 3px rgba(0,0,0,0.08)}header a{color:black}article,footer{padding:1.25em 1.5em 1.5em 1.5em;margin:0 auto}article{max-width:840px}footer{clear:both;border-top:solid 1px #c8ccd1;background-color:#eaecf0}.actions{display:flex;justify-content:space-around;margin-bottom:1em}.actions a{background-color:#f8f9fa;color:#222;border-color:#a2a9b1;border-style:solid;border-width:1px;border-radius:2px;padding:6px 12px;position:relative;min-height:2.28571429em;font-weight:bold;text-decoration:none;vertical-align:top;text-align:center;cursor:pointer;display:inline-block;-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box;font-family:inherit;font-size:inherit;white-space:nowrap;-webkit-touch-callout:none;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}h1{line-height:1.3;word-wrap:break-word;word-break:break-word;font-family:'Linux Libertine','Georgia','Times',serif;font-size:1.7em;font-weight:inherit;margin:0 0 12px;padding:0;border:0;vertical-align:baseline;background:none}.tabs{border-bottom:1px solid #eaecf0 }.tabs a{font-size:0.85em;margin:0 10px 0 0;color:#54595d;font-weight:bold;padding-bottom:6px;display:inline-block;text-decoration:none}.tabs a.selected{border-bottom:2px solid #54595d}.thumb.tright{float:right;clear:right;margin:0.5em 0 1.3em 1.4em }.wikitable{background-color:#f8f9fa;color:#222;margin:1em 0;border:1px solid #a2a9b1;border-collapse:collapse}.wikitable td{border:1px solid #a2a9b1;padding:0.2em 0.4em}table{display:block;width:100%;overflow:auto;overflow-y:hidden}a{color:#0645ad;text-decoration:none}.site-links{margin-top:1em;display:flex;justify-content:space-between}
    </style>
    <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>

    <link rel=\\"canonical\\" href=\\"http://localhost:8080/w/index.php/UTPage\\">
    <title>UTPage</title>
</head>
<body>
<amp-auto-ads type=\\"adsense\\" data-ad-client=\\"ca-pub-4622825295514928\\"></amp-auto-ads>
<amp-analytics type=\\"gtag\\" data-credentials=\\"include\\">
    <script type=\\"application/json\\">
		{
			\\"vars\\" : {
				\\"gtag_id\\": \\"UA-21207284-8\\",
				\\"config\\" : {
					\\"UA-21207284-8\\": { \\"groups\\": \\"default\\" }
				}
			}
		}
    </script>
</amp-analytics>
<header>
    <a href=\\"/w/index.php/Main_Page\\">devwiki</a>
    <a href=\\"/w/index.php/Main_Page\\">
        <amp-img alt=\\"logo\\" width=\\"48px\\" height=\\"48px\\" src=\\"https://www.droidwiki.org/static/images/project-logos/androide.png\\"></amp-img>
    </a>
</header>
<article>
    <h1>UTPage</h1>
    <div class=\\"tabs\\">
        <a class=\\"selected\\" href=\\"/w/index.php/UTPage\\">Page</a>
        <a href=\\"/w/index.php/Talk:UTPage\\">Discussion</a>
        <a href=\\"/w/index.php?title=UTPage&amp;action=history\\">Page history</a>
    </div>
    <div class=\\"mw-parser-output\\"><div class=\\"mw-parser-output\\"><p>UTContent
</p>


</div>
<form target=\\"_blank\\" name=\\"searchbox\\" action=\\"/wiki/Special:Search\\">
<input name=\\"search\\" type=\\"text\\" value=\\"\\" />
<input type=\\"submit\\" name=\\"go\\" class=\\"mw-ui-button\\" value=\\"Search\\" />
</form>
</div>
    <div class=\\"actions\\">
        <a href=\\"/w/index.php?title=UTPage&amp;action=edit\\">Edit</a>
    </div>
	<div class=\\"catlinks\\">Categories: <a href=\\"/w/index.php/Category:Test-Category\\" title=\\"Category:Test-Category\\">Test-Category</a></div>
</article>
<footer>
    <div>Content is available under CC test unless otherwise noted.</div>
    <div class=\\"site-links\\">
        <span><a href=\\"/w/index.php/Devwiki:About\\" title=\\"Devwiki:About\\">About devwiki</a></span>
		<span><a href=\\"/w/index.php/Devwiki:General_disclaimer\\" title=\\"Devwiki:General disclaimer\\">Disclaimers</a></span>
		
    </div>
</footer>
</body>
</html>
"
`;
